### PR TITLE
feat!: make `$rename()`'s behavior closer to python polars'

### PR DESCRIPTION
**Pull request by eitsupi**
A follow up for #1122

- `old_name = "new_name"` instead of `new_name = "old_name"`
- Allow a function input

### CODE REVIEW

**Comment by eitsupi** [comment]
@etiennebacher This will definitely break tidypolars, but hopefully it is a meaningful change in terms of consistency with Python.

File: R/lazyframe__lazy.R [hunk 1]
```diff
@@ -1540,66 +1540,46 @@ LazyFrame_melt = function(
...
   mapping = list2(...)
   if (length(mapping) == 0) {
     return(self)
+  } else if (is.function(mapping[[1L]])) {
+    {
+      existing = names(self)
+      new = mapping[[1L]](existing)
+    } |>
+      result() |>
+      uw()
```
Comment (etiennebacher) [inline R/lazyframe__lazy.R:1571]: Nitpick but I find 

```r
result({
 existing = names(self)
 new = mapping[[1L]](existing)
}) |>
  uw()
```
easier to read

**Review by etiennebacher — APPROVED** [review]
Nice, I prefer this order and actually I originally put this `old = new` order when I implemented it one year ago but we decided to follow `dplyr` for more consistency (https://github.com/pola-rs/r-polars/pull/239). 

> This will definitely break tidypolars

Not a big deal, I'll just need to update the internals but it won't change the external behaviour. 

Also for those who use `polars` directly, this will clearly error and not just fail silently.

**Comment by eitsupi** [comment]
> Nice, I prefer this order and actually I originally put this `old = new` order when I implemented it one year ago but we decided to follow `dplyr` for more consistency (#239).

Hmmm, I completely forgot about this.
And I always have trouble with the order of `dplyr::rename()` as well (although `new=old` makes more sense since it is consistent with `dplyr::select()`)

**Pull request merged**